Arsenal.gd

```diff
@@ -21,24 +21,24 @@ func _ready():
 func _process(delta):
 	if Input.is_action_pressed("fire_primary"):
 		if primary_weapon and primary_weapon.has_method("firing"):
-			primary_weapon.firing()
+			primary_weapon.firing(get_parent())
```

Comment:
passing in the source of projectiles here so we can not collide with the parent

i do this because enemy ships fire the same projectile scene

we could have a separate "enemy harpoon" scene if we dont like this way though